src/d3a/d3a_core/sim_results/device_statistics.py

```diff
@@ -166,4 +164,4 @@ def _gather_device_statistics(cls, area: Area, subdict: Dict, flat_result_dict:
             _create_or_append_dict(subdict, "production_kWh",
                                    {market.time_slot: area.strategy.energy_per_slot_kWh})
 
-        flat_result_dict[area.uuid] = deepcopy(subdict)
+        flat_result_dict[area.uuid] = subdict.copy()
```

Comment:
And this makes a significant difference?

Reply:
Indeed it does. As we discussed in the former comment, shallow copy is much much faster than deepcopy.